samples/app-ReporterPlus/ReporterPlus/Helpers/BlobHelper.cs

```diff
@@ -0,0 +1,160 @@
+            }
+            else
+            {
+                submittedByUserImageUrl = Constants.MemberGenericImageUrl;
+            }
+
+            if (taskModuleOutput.data.ImagesOutput.Length != 0)
+            {
+                foreach (string item in taskModuleOutput.data.ImagesOutput)
+                {
+                    imageURL.Add(item);
+                }
+            }
+            else
+            {
+                imageURL = null;
+            }
+
+            if(imageURL!= null){
+                foreach (var item in imageURL)
+                {
+                    var imgs = new Images { url = item };
+                    imgURL.Add(imgs);
+                }
+            }
+
+            string[] blobStorage = { ".", "Resources", "BlobStorage.json" };
+            var blobStorageText = File.ReadAllText(Path.Combine(blobStorage));
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(blobStorageText);
+
+            var payloadData = new
+            {
+                requestID = requestID,
+                status = "Pending",
+                itemName = taskModuleOutput.data.ItemName,
+                itemCode = taskModuleOutput.data.ItemCode,
+                assignedToName = taskModuleOutput.data.AssignedTo.displayName,
+                assignedToId = taskModuleOutput.data.AssignedTo.objectId,
+                assignedToMail = taskModuleOutput.data.AssignedTo.email,
+                submittedByName = turnContext.Activity.From.Name,
+                submittedById = turnContext.Activity.From.AadObjectId,
+                submittedByMail = taskModuleOutput.data.SubmittedByMail,
+                assignedToUserImage = assignedToUserImageUrl,
+                submittedByUserImage = submittedByUserImageUrl,
+                imageURL = imgURL,
+                audioURL = audioURL,
+                comments = taskModuleOutput.data.Comments,
+                conversationId = turnContext.Activity.Conversation.Id,
+                messageId = "",
+            };
+
+            var blobTextString = template.Expand(payloadData);
+            streamConverter = Encoding.ASCII.GetBytes(blobTextString);
+            blob = container.GetBlobClient(requestID + ".txt");
+            await blob.UploadAsync(new MemoryStream(streamConverter));
+            return requestID;
+        }
+
+        public static async Task<BlobDataDeserializer> GetBlob(string reqId, string request, string messageId = null)
+        {
+            var fileName = reqId + ".txt";
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(Constants.BlobConnectionString);
+            CloudBlobClient serviceClient = storageAccount.CreateCloudBlobClient();
+            CloudBlobContainer container = serviceClient.GetContainerReference(Constants.BlobContainerName);
+            CloudBlockBlob blob = container.GetBlockBlobReference(fileName);
+            if (request == "Approved" || request == "Rejected" || messageId != null)
```

Comment:
Please check out my suggestion the ReporterPlusBot.cs and try to update these cases to use the new class or enum and also update string comparisons.